pallets/handles/src/handles_signed_extension.rs

```diff
@@ -45,8 +46,18 @@ impl<T: Config + Send + Sync> HandlesSignedExtension<T> {
 		let delegator_msa_id = T::MsaInfoProvider::ensure_valid_msa_key(&delegator_key)
 			.map_err(|e| map_dispatch_error(e))?;
 		// Validation: The MSA must already have a handle associated with it
-		_ = MSAIdToDisplayName::<T>::try_get(delegator_msa_id)
+		let handle_from_state = MSAIdToDisplayName::<T>::try_get(delegator_msa_id)
 			.map_err(|_| UnknownTransaction::CannotLookup)?;
+		let expiration = handle_from_state.1;
+		let current_block = frame_system::Pallet::<T>::block_number();
+
+		let is_past_min_lifetime = current_block >= expiration;
+		ensure!(
+			is_past_min_lifetime,
+			map_dispatch_error(DispatchError::Other(
+				Error::<T>::HandleWithinMortalityPeriod.into()
+			))
```

Comment:
@wilwade  i believe it is safe to also check for expiration in signed extension? I have moved that check here